engine/src/main/java/org/terasology/world/chunks/internal/ChunkImpl.java

```diff
@@ -64,6 +65,9 @@
     private static final DecimalFormat PERCENT_FORMAT = new DecimalFormat("0.##");
     private static final DecimalFormat SIZE_FORMAT = new DecimalFormat("#,###");
 
+    private final int verticalChunkMeshSegments = CoreRegistry.get(Config.class).getSystem().getVerticalChunkMeshSegments();
+    private final int chunkHalfHeight = ChunkConstants.SIZE_Y / verticalChunkMeshSegments / 2;
```

Comment:
These two lines, partially addressing issue #1536, favor performance in place of memory consumption, as by my understanding they add 8 bytes to each chunk. At the Ultra setting this is about 100KB more memory used.

Ideally they should be static, but then we end up in the same problem we had in ChunkConstants, with errors resulting from the Config not being available yet. 

Alternatively I can inline everything in exchange of a small performance hit.